schema-registry/serdes/src/main/java/com/hortonworks/registries/schemaregistry/serdes/avro/AvroSnapshotSerializer.java

```diff
@@ -42,38 +40,17 @@
  */
 public class AvroSnapshotSerializer extends AbstractAvroSnapshotSerializer<byte[]> {
 
-    /**
-     * Property name for protocol version to be set with {@link #init(Map)}.
-     */
-    public static final String SERDES_PROTOCOL_VERSION = "serdes.protocol.version";
-
-    private Byte protocolVersion;
-
     public AvroSnapshotSerializer() {
     }
 
     public AvroSnapshotSerializer(ISchemaRegistryClient schemaRegistryClient) {
         super(schemaRegistryClient);
     }
 
-    @Override
-    public void init(Map<String, ?> config) {
```

Comment:
The protocol changing or support should be fundamental, and also allow re-use in other concrete impls MessageContextBasedAvroDeserializer. e.g. avoids having to declare/hardcode the DefaultAvroSerDesHandler in there, instead now other concrete's like MessageContextBasedAvroDeserializer can support different payload avro serialisation protocols.